web/src/components/welcome-modal/welcome-modal.tsx

```diff
@@ -30,21 +33,44 @@ export default ({ challengeToken, teamToken, ...props }: WelcomeModalProps) => {
 
   useEffect(() => trackChallenge('modal-welcome'), []);
 
-  const parseEnrollment = (queryString: string): Enrollment => {
-    const regex = new RegExp(/([a-z]+)=([a-z]+)/, 'gm');
+  const parseEnrollment = (
+    queryString: string,
+    referer?: string
+  ): Enrollment => {
+    const regex = new RegExp(/([\w\-]+)=([\w\-]+)/, 'g');
```

Comment:
The `m` was a leftover value from when I was testing the regex so it's been taken out. This pattern works for something like `?key_name=value-PROPERTY-123` which I think covers all of our current use cases.